### DISCUSSION TITLE
[R] duckdb version of `dbplyr::tbl_memdb()` ? (or `arrow::to_duckdb()` without `arrow`)

### DISCUSSION THREAD

**Discussion by eitsupi** [duckdb/duckdb#4041] [Ideas]
We can register a data frame in SQLite and query its tables with the following pipeline.

```r
mtcars |> dbplyr::tbl_memdb() |> dplyr::select(cyl)
```

We can do something similar with duckdb, but I think we need the `arrow` package for that.

```r
mtcars |> arrow::to_duckdb() |> dplyr::select(cyl)
```

If a function like this `to_duckdb` were available in the `duckdb` package, it might be useful to be able to easily write dplyr queries using duckdb as a backend without installing the `arrow` package.

**Comment by hannes** [comment]
@krlmlr maybe we can do something here with the register functionality?

**Comment by krlmlr** [comment]
Or in tidyduck? Should we aim for a first CRAN release soon-ish?
